api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/GraphProcess.scala

```diff
@@ -0,0 +1,28 @@
+package pl.touk.nussknacker.engine.api.deployment
+
+import io.circe.parser.parse
+import io.circe.{Decoder, Encoder, Json}
+
+object GraphProcess {
+
+  implicit val encoder: Encoder[GraphProcess] = Encoder.encodeJson.contramap(_.json)
+  implicit val decoder: Decoder[GraphProcess] = Decoder.decodeJson.map(GraphProcess(_))
+
+  val empty: GraphProcess = GraphProcess("{}")
+
+  //TODO: Add json validation here?
+  def apply(jsonString: String): GraphProcess = {
+    val json = parse(jsonString) match {
+      case Left(_) => throw new IllegalArgumentException(s"Invalid raw json string: $jsonString.")
+      case Right(json) => json
+    }
+
+    new GraphProcess(json)
+  }
+
+}
+
+final case class GraphProcess private(json: Json) extends AnyVal {
+  override def toString: String = jsonString
+  def jsonString: String = json.spaces2
```

Comment:
don't think it's needed here, we should use it rarely - only when serializing stuff during deployment..

Reply:
I disagree, earlier we had a lot of usage: space2 / noSpaces on different places. Beacuse of that in db we have sometimes json with spacees or json without spaces. I think better way is normalize it and decide how present json in string in only one way.

Reply:
I've removed jsonString, right nowe is only toString method.